client/packages/invoices/src/Prescriptions/DetailView/columns.ts

```diff
@@ -202,28 +205,84 @@ export const usePrescriptionColumn = ({
           ]),
       },
     ],
-    [
-      'unitQuantity',
-      {
-        accessor: ({ rowData }) => {
-          if ('lines' in rowData) {
-            const { lines } = rowData;
-            return ArrayUtils.getUnitQuantity(lines);
+  ];
+
+  if (newPrefs?.displayVaccineInDoses) {
+    columns.push({
+      key: 'doses',
+      label: 'label.doses-per-unit',
+      sortable: false,
+      accessor: ({ rowData }) => {
+        if ('lines' in rowData) {
+          const { lines } = rowData;
+          if (lines[0]?.item.isVaccine) {
+            const doses = lines?.map(
+              ({ item }) => item?.doses ?? UNDEFINED_STRING_VALUE
+            );
+            const dosesTheSame = doses?.every(dose => dose === doses?.[0]);
+            return dosesTheSame ? doses?.[0] : t('multiple');
           } else {
-            return rowData.packSize * rowData.numberOfPacks;
+            return UNDEFINED_STRING_VALUE;
           }
-        },
-        getSortValue: rowData => {
-          if ('lines' in rowData) {
-            const { lines } = rowData;
-            return ArrayUtils.getUnitQuantity(lines);
-          } else {
-            return rowData.packSize * rowData.numberOfPacks;
+        } else {
+          return rowData?.item?.isVaccine
+            ? (rowData?.item?.doses ?? UNDEFINED_STRING_VALUE)
+            : UNDEFINED_STRING_VALUE;
+        }
+      },
+    });
+  }
+  columns.push([
+    'unitQuantity',
+    {
+      accessor: ({ rowData }) => {
+        if ('lines' in rowData) {
+          const { lines } = rowData;
+          const displayVaccineInDoses =
+            newPrefs?.displayVaccineInDoses && lines[0]?.item.isVaccine;
+          const unitQuantity = ArrayUtils.getUnitQuantity(lines);
+
+          // TODO: Different doses with item variants
+          if (displayVaccineInDoses) {
+            const doses = lines[0]?.item.doses ?? 1;
+            return unitQuantity * doses;
```

Comment:
Oh now I am confused - different to inbounds, and you're showing num of doses, but in the unit quantity column? If unit is vials, and I have 10 doses per vial, I want to be seeing either 0.1 units (0.1 vials) or 1 dose, but I don't think I should see 1 unit here...

Should be 0.1 for unit and pack?
![Screenshot 2025-05-05 at 11 25 18 AM](https://github.com/user-attachments/assets/bb62ebbd-93ee-4d80-b0af-706f25666978)

Given:
![Screenshot 2025-05-05 at 11 24 58 AM](https://github.com/user-attachments/assets/fe0d0275-1a9c-4623-803b-728f7424a4b9)

Reply:
I think we should define sep columns rather than try reuse

Reply:
Prescriptions is different to inbounds because when you dispense you're not dispensing it to the patient in vials.

We can't guarantee that they will ONLY be dispensing vaccine items... kinda of hard to use separate columns, are you saying that we should add both columns and have unit qty show `-` for dose items? And then have a dose qty that shows `-` for non-vaccine?

Reply:
Yeah - or I suppose no harm in unit quantity being 0.1 for the dose items? But yes one of those two - we shouldn't call unit "dose"